### PR TITLE
Get and Post data

### PR DESCRIPTION
**Pull request by vincentmigot**

### CODE REVIEW

File: phis2-ws/src/main/java/phis2ws/service/configuration/DateFormat.java [hunk 1]
```diff
@@ -8,6 +8,13 @@
 package phis2ws.service.configuration;
 
 //SILEX:todo
```
Comment (annetireau) [inline phis2-ws/src/main/java/phis2ws/service/configuration/DateFormat.java]: could you group //SILEX:todo target

File: phis2-ws/src/main/java/phis2ws/service/dao/mongo/DataDAOMongo.java [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+import phis2ws.service.documentation.StatusCodeMsg;
+import phis2ws.service.ontologies.Oeso;
+import phis2ws.service.utils.POSTResultsReturn;
+import phis2ws.service.utils.UriGenerator;
+import phis2ws.service.view.brapi.Status;
+import phis2ws.service.view.model.phis.Data;
+
+/**
+ * Represents the MongoDB Data Access Object.
+ */
```
Comment (annetireau) [inline phis2-ws/src/main/java/phis2ws/service/dao/mongo/DataDAOMongo.java]: Add @author

File: phis2-ws/src/main/java/phis2ws/service/dao/mongo/DataDAOMongo.java [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+                
+                // Add data to the list
+                dataList.add(data);
+            }
+        }
+        
+        return dataList;
+    }
+
+    @Override
```
Comment (annetireau) [inline phis2-ws/src/main/java/phis2ws/service/dao/mongo/DataDAOMongo.java]: Please add example and comment

File: phis2-ws/src/main/java/phis2ws/service/dao/mongo/DataDAOMongo.java [hunk 4]
```diff
@@ -0,0 +1,396 @@
...
+        }
+        
+        query.append(DB_FIELD_VARIABLE, variableUri);
+        
+        LOGGER.debug(getTraceabilityLogs() + " query : " + query.toString());
+        
+        return query;
+    }
+
+    public int count() {
```
Comment (annetireau) [inline phis2-ws/src/main/java/phis2ws/service/dao/mongo/DataDAOMongo.java]: Add comment, @param, etc.

File: phis2-ws/src/main/java/phis2ws/service/resources/DataResourceService.java [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+                return Response.status(Response.Status.BAD_REQUEST).entity(postResponse).build();
+            }       
+        } catch (ParseException e) {
+            postResponse = new ResponseFormPOST(new Status(StatusCodeMsg.REQUEST_ERROR, StatusCodeMsg.ERR, e.getMessage()));
+            return Response.status(Response.Status.BAD_REQUEST).entity(postResponse).build();
+        }
+    }
+    
+    /**
+     * Generayes an data list from a given list of DataPostDTO.
```
Comment (annetireau) [inline phis2-ws/src/main/java/phis2ws/service/resources/DataResourceService.java]:
```suggestion
     * Generates an data list from a given list of DataPostDTO.
```

File: phis2-ws/src/main/java/phis2ws/service/resources/DataResourceService.java [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+import phis2ws.service.resources.validation.interfaces.URL;
+import phis2ws.service.utils.POSTResultsReturn;
+import phis2ws.service.view.brapi.Status;
+import phis2ws.service.view.brapi.form.AbstractResultForm;
+import phis2ws.service.view.brapi.form.ResponseFormData;
+import phis2ws.service.view.brapi.form.ResponseFormPOST;
+import phis2ws.service.view.model.phis.Data;
+
+/**
+ * Data resource service
```
Comment (annetireau) [inline phis2-ws/src/main/java/phis2ws/service/resources/DataResourceService.java]: Add @author

File: phis2-ws/src/main/java/phis2ws/service/resources/DataResourceService.java [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+        ArrayList<Data> dataList = new ArrayList<>();
+        
+        for (DataPostDTO dataDTO : dataDTOs) {
+            dataList.add(dataDTO.createObjectFromDTOWithException());            
+        }
+        
+        return dataList;
+    }
+
+    @GET
```
Comment (annetireau) [inline phis2-ws/src/main/java/phis2ws/service/resources/DataResourceService.java]: Add @example, @param, etc.

File: phis2-ws/src/main/java/phis2ws/service/configuration/DateFormat.java (hunk 1 above)
Comment (vincentmigot) [reply phis2-ws/src/main/java/phis2ws/service/configuration/DateFormat.java]: Ok

File: phis2-ws/src/main/java/phis2ws/service/dao/mongo/DataDAOMongo.java (hunk 2 above)
Comment (vincentmigot) [reply phis2-ws/src/main/java/phis2ws/service/dao/mongo/DataDAOMongo.java]: Ok

File: phis2-ws/src/main/java/phis2ws/service/dao/mongo/DataDAOMongo.java (hunk 4 above)
Comment (vincentmigot) [reply phis2-ws/src/main/java/phis2ws/service/dao/mongo/DataDAOMongo.java]: Ok

File: phis2-ws/src/main/java/phis2ws/service/resources/DataResourceService.java (hunk 5 above)
Comment (vincentmigot) [reply phis2-ws/src/main/java/phis2ws/service/resources/DataResourceService.java]: Ok

File: phis2-ws/src/main/java/phis2ws/service/resources/DataResourceService.java (hunk 6 above)
Comment (vincentmigot) [reply phis2-ws/src/main/java/phis2ws/service/resources/DataResourceService.java]: Ok

File: phis2-ws/src/main/java/phis2ws/service/resources/DataResourceService.java (hunk 7 above)
Comment (vincentmigot) [reply phis2-ws/src/main/java/phis2ws/service/resources/DataResourceService.java]: Ok

**Comment by vincentmigot** [comment]
Corrections done and bug fixed when inserting data (Deployed on test server)

File: phis2-ws/src/main/java/phis2ws/service/dao/mongo/DataDAOMongo.java [hunk 8]
```diff
@@ -0,0 +1,413 @@
...
+
+        for (Data data : dataList) {
+            //1. Check if the variableUri exist and is a variable
+            if (!variableDAO.existAndIsVariable(data.getVariableUri())) {
+                dataOk = false;
+                checkStatus.add(new Status(StatusCodeMsg.WRONG_VALUE, StatusCodeMsg.ERR,
+                        "Unknwon variable : " + data.getVariableUri()));
+
+               
+            } else if (!variableDAO.existUri(data.getObjectUri())) {
```
Comment (annetireau) [inline phis2-ws/src/main/java/phis2ws/service/dao/mongo/DataDAOMongo.java]: Add comment and provenance test !

**Pull request merged**